gnomad_qc/v4/create_release/calculate_variant_statistics.py

```diff
@@ -33,156 +27,100 @@
-            ~hl.is_missing(ht_release_data[vds.variant_data.row_key])
-        )
-        .rows()
-        .select(),
-    )
+    # Read in release data and select relevant annotations.
+    ht = release_sites(data_type=data_type).ht()
+    ht = ht.select("freq", "filters", "region_flags", "vep")
 
-    vds = hl.vds.filter_samples(
-        vds,
-        vds.reference_data.filter_cols(meta_ht[vds.reference_data.s].release)
-        .cols()
-        .select(),
-    )
+    if test:
+        logger.info("Test: filtering to variants on chr22")
+        ht = ht.filter(ht.locus.contig == "chr22")
 
-    # Filter to variants only in requested calling intervals.
-    if filter_ukb_regions:
-        logger.info("Filter to only variants in UKB capture regions.")
-        filtered_ukb = (
-            vds.variant_data.filter_rows(
-                ht_release_data[
-                    vds.variant_data.row_key
-                ].region_flags.outside_ukb_capture_region
-            )
-            .rows()
-            .select()
-        )
-        vds = hl.vds.filter_variants(vds, filtered_ukb)
-
-    if filter_broad_regions:
-        logger.info("Filter to only variants in Broad capture regions.")
-        filtered_broad = (
-            vds.variant_data.filter_rows(
-                ht_release_data[
-                    vds.variant_data.row_key
-                ].region_flags.outside_broad_capture_region
-            )
-            .rows()
-            .select()
-        )
-        vds = hl.vds.filter_variants(vds, filtered_broad)
+    logger.info("Filter VDS to variants in gnomAD release.")
+    vds = hl.vds.filter_variants(vds, ht.select())
 
-    # Create variant matrix table from VDS and annotate with all relevant info
+    # Create variant matrix table from VDS and annotate with all relevant info.
     vmt = vds.variant_data
-    logger.info("VDS created and checkpointing, without annotations")
+    logger.info("Variant MT created and checkpointing, without annotations")
     vmt = vmt.checkpoint(
-        new_temp_file(f'vmt_{"TEST" if test else ""}_mt', extension="mt"),
+        new_temp_file(f'vmt_{"TEST" if test else ""}', extension="mt"),
         overwrite=overwrite,
     )
 
-    # Add extra Allele Count and A-Type, according to Hail standards, to help their computation
+    # Add extra Allele Count and Allele Type annotations to variant Matrix
+    # Table, according to Hail standards, to help their computation.
     ac_and_atype = vmt_sample_qc_variant_annotations(
         global_gt=vmt.GT, alleles=vmt.alleles
     )
     vmt = vmt.annotate_rows(**ac_and_atype)
 
-    # NOTE: not size efficient, but we are never checkpointing THIS I don't believe
-    vmt = vmt.annotate_rows(**ht_release_data[vmt.row_key])
-
-    if agg_lof:
-        logger.info("Looking at variants that are VEP LOF")
-        vmt_vep = vep.filter_vep_transcript_csqs(
-            vmt,
+    if by_csqs:
+        ht = filter_vep_transcript_csqs(
```

Comment:
Did you or Julia agree on filtering to only MANE select transcripts? I leave it the same for now.